controls/raddiagram/populating-with-data/data-databinding.md

```diff
@@ -372,29 +372,31 @@ Now let's take care of our Shapes and Connections' look and feel. We have to bin
 
 The final step is to bind the View to the ViewModels:
 		
-#### __C#__	
-{{region raddiagram-data-databinding-11}}
+#### __[C#] Example 7: Setting the GraphSource__	
+{{region cs-raddiagram-data-databinding-11}}
 	this.diagram.GraphSource = CreatePascalTriangleGraphSource(7);
 {{endregion}}
 
-#### __VB.NET__
-{{region raddiagram-data-databinding-12}}
+#### __[VB.NET] Example 7: Setting the GraphSource__
+{{region vb-raddiagram-data-databinding-12}}
 	Me.diagram.GraphSource = CreatePascalTriangleGraphSource(7)
 {{endregion}}
 
-Below you can see the result - __Pascal Triangle__ with seven levels represented by __RadDiagram__:
-![raddiagrams populating with data data binding](images/raddiagrams_populating_with_data_data_binding.png)
+#### __Figure 1: **Pascal Triangle** with seven levels represented by **RadDiagram**__
+![Pascal Triangle with seven levels represented by RadDiagram](images/raddiagrams_populating_with_data_data_binding.png)
 
 ## Two-Way MVVM
 
 In Q2 2012 we improved the MVVM support by adding two-way changes, i.e. changes in the UI or diagram will be reflected in the viewmodel. 
 
 In order to make use of the two-way binding the data source should implement the __IObservableGraphSource__ interface which extends the __IGraphSource__ interface with some methods called by __RadDiagram__ when Items are added or removed in the diagram.		
 
-The simplest way to use the __IObservableGraphSource__ is via its default implementation in the Extensions library. The __ObservableGraphSourceBase__ contains Add and Remove methods which can and should be overriden to make sure that any changes in the view (__RadDiagram__ canvas), will be reflected in the ViewModel. Let's take a simple example. Define first a data entity called Person:
+The simplest way to use the __IObservableGraphSource__ is via its default implementation in the Extensions library. The __ObservableGraphSourceBase__ contains Add and Remove methods which can and should be overriden to make sure that any changes in the view (__RadDiagram__ canvas), will be reflected in the ViewModel. The next examples demonstrate this in a sample scenario.
+
+> Note that, when an item (node) is removed from the GraphSource, its corresponding UI container (the RadDiagramShape) is stored in a collection of 'recycled' shapes for future use. This aims to speed up the diagram performance in extensive undo-redo and container generation operations. However, this could lead to an increased use of memory. In order to clear this cache, utilize the __ClearCache__ method of the diagram's __ContainerGenerator__ like so: "(this.Diagram.ContainerGenerator as GenericContainerGenerator<Telerik.Windows.Controls.Diagrams.RadDiagramItem>).ClearCache();".

```

Comment:
I think we should extract the ClearCache invocation in a separate code snippet so that it is more visible. Maybe we can have this note converted to a standard paragraph for the purpose.